server/src/main/java/org/apache/druid/server/coordinator/duty/KillUnusedSegments.java

```diff
@@ -159,10 +176,9 @@ DruidCoordinatorRuntimeParams runInternal(DruidCoordinatorRuntimeParams params)
         dataSourcesToKill = segmentsMetadataManager.retrieveAllDataSourceNames();
       }
 
-      log.debug("Killing unused segments in datasources: %s", dataSourcesToKill);
-      lastKillTime = System.currentTimeMillis();
+      log.debug("Killing unused segments for datasources[%s]", dataSourcesToKill);
+      lastKillTime = DateTimes.nowUtc();
```

Comment:
Nit: Wondering if we should update `lastKillTime` after the kill task has been successfully submitted?

It would have the benefit of not updating `lastKillTime` if there was an error in submitting the kill task. Thus, the duty would retry submit in the next coordinator run.
Drawback is that if there is some bug and submitting a kill task failed once, this duty would keep trying to submit a kill task in every coordinator run and failing, flooding the log with errors. I guess that's okay?

Reply:
Yeah, I think this makes sense. Would defer making this behavioral change in a follow-up to avoid mixing up too many unrelated changes in this patch